### PR TITLE
🧪 Add unit tests for SearchForm component

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** Added missing unit tests for the `SearchForm.tsx` component in the `web` package to verify form interaction, submission behavior, and disabled state handling.

📊 **Coverage:** The new test suite covers:
- Correct initial rendering of inputs and buttons
- Successful submission, verifying the `onSearch` prop is called with the correct keyword query and max results
- Edge cases preventing submission when the query is empty or consists only of whitespace
- UI constraints ensuring the submit button is disabled when the keyword input is empty
- Visual states and constraints ensuring inputs and the submit button are disabled when the `loading` prop is `true`

✨ **Result:** Improved test coverage for the core search UI and prevented future regressions in form behavior. All tests pass locally without regressions.

---
*PR created automatically by Jules for task [18180329870528769746](https://jules.google.com/task/18180329870528769746) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

`SearchForm.tsx` コンポーネントに対するユニットテストを新規追加するPRです。レンダリング・送信・無効化状態など主要なケースはカバーされており、テスト自体は正常に動作します。
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

P2のみのため、マージは安全です。

テストはすべて正常に動作し、主要なUIの振る舞いをカバーしています。指摘はいずれもP2（スタイル・カバレッジの改善提案）であり、バグや動作の誤りはありません。

packages/web/src/components/SearchForm.test.tsx — handleSubmitの空白ガード検証とtrim()の動作確認を補強するとテスト品質が向上します。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web/src/components/SearchForm.test.tsx | SearchFormコンポーネントのユニットテストを新規追加。主要な動作はカバーされているが、handleSubmitの空白ガードが未検証、trim()の効果が確認できていないなどP2の改善点あり。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant SearchForm
    participant handleSubmit
    participant onSearch

    User->>SearchForm: キーワード入力 (setQuery)
    User->>SearchForm: Max Results 変更 (setMaxResults)
    User->>SearchForm: Search ボタンクリック
    SearchForm->>handleSubmit: onSubmit イベント
    handleSubmit->>handleSubmit: e.preventDefault()
    alt query.trim() が truthy
        handleSubmit->>onSearch: onSearch(query.trim(), maxResults)
    else query が空 / 空白のみ
        handleSubmit-->>SearchForm: 何もしない
    end

    Note over SearchForm: loading=true の場合<br/>inputs・button を disabled に設定
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 3 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 3
packages/web/src/components/SearchForm.test.tsx:38-45
**空白クエリのガード検証が不十分**

`handleSubmit` 内の `if (query.trim())` ガードは全くテストされていません。クエリが空白文字列 `"   "` のとき、ボタンの `disabled={!query.trim()}` も `true` になるため、`fireEvent.click` は jsdom でフォーム送信をトリガーしません。つまり `onSearchMock` が呼ばれないのは「ハンドラが弾いたから」ではなく「ボタンが無効だから」です。`fireEvent.submit(formElement)` でフォームを直接送信することで、ボタンの `disabled` 状態を迂回して `handleSubmit` のガードそのものを検証できます。

### Issue 2 of 3
packages/web/src/components/SearchForm.test.tsx:19-29
**`query.trim()` の動作が検証されていない**

`SearchForm.tsx` の `handleSubmit` では `onSearch(query.trim(), maxResults)` を呼び出しており、先頭・末尾の空白が除去された文字列が渡されます。しかし、テストで入力される `"test query"` は前後に空白がないため、`trim()` の有無に関わらず結果が同じです。`" test query "` など前後にスペースを含む値を入力し、`onSearchMock` が `"test query"` (trimmed) で呼ばれることを確認するテストを追加すると、`trim()` の効果を直接検証できます。

### Issue 3 of 3
packages/web/src/components/SearchForm.test.tsx:1-4
**`@testing-library/user-event` が未導入**

`fireEvent` は DOM イベントを直接発火させるため、実際のユーザー操作（フォーカス移動・入力ディレイ等）を再現できません。`@testing-library/user-event` の `userEvent.type()` / `userEvent.click()` を使うとよりリアルなインタラクションが検証できます。ただし `package.json` の `devDependencies` に `@testing-library/user-event` が含まれていないため、導入が必要です。

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test(web): add SearchForm component test..."](https://github.com/hiroki-org/paper-tools/commit/e6bd2564b3e22ed07885cef9e72f40d46b9ede5a) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30577618)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: packages/web/src/components/SearchForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        fireEvent.change(keywordInput, { target: { value: "test query" } });
+
+        const maxResultsInput = screen.getByLabelText("Max Results");
+        fireEvent.change(maxResultsInput, { target: { value: "50" } });
+
+        const submitButton = screen.getByRole("button", { name: "Search" });
+        fireEvent.click(submitButton);
+
+        expect(onSearchMock).toHaveBeenCalledTimes(1);
+        expect(onSearchMock).toHaveBeenCalledWith("test query", 50);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/components/SearchForm.test.tsx:32]: 対応しました。通常送信テストで前後に空白を含むクエリを userEvent で入力し、onSearchMock が trimmed な "test query" と maxResults 50 で呼ばれることを検証するようにしました。検証: pnpm --filter @paper-tools/web test -- src/components/SearchForm.test.tsx / git diff --check

File: packages/web/src/components/SearchForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        const submitButton = screen.getByRole("button", { name: "Search" });
+
+        // Empty query (initial state)
+        fireEvent.click(submitButton);
+        expect(onSearchMock).not.toHaveBeenCalled();
+
+        // Whitespace query
+        const keywordInput = screen.getByLabelText("Keyword");
+        fireEvent.change(keywordInput, { target: { value: "   " } });
+        fireEvent.click(submitButton);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/components/SearchForm.test.tsx]: 対応しました。空白のみ入力のケースは disabled なボタンのクリックではなく、formElement に対する fireEvent.submit で直接送信し、handleSubmit 内の query.trim() ガード自体を検証するように変更しました。検証: pnpm --filter @paper-tools/web test -- src/components/SearchForm.test.tsx / git diff --check

File: packages/web/src/components/SearchForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// @vitest-environment jsdom
+import { describe, it, expect, vi } from "vitest";
+import { render, screen, fireEvent } from "@testing-library/react";
+import SearchForm from "./SearchForm";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/components/SearchForm.test.tsx:5]: 対応しました。@testing-library/user-event を packages/web の devDependency に追加し、通常の入力・クリック操作を userEvent に置き換えました。空白ガード検証のみ、指摘どおり form submit を直接発火するため fireEvent.submit を残しています。検証: pnpm --filter @paper-tools/web test -- src/components/SearchForm.test.tsx / git diff --check

File: packages/web/src/components/SearchForm.test.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+describe("SearchForm", () => {
+    it("renders correctly with default props", () => {
+        render(<SearchForm onSearch={vi.fn()} />);
+
+        expect(screen.getByLabelText("Keyword")).toBeTruthy();
+        expect(screen.getByLabelText("Max Results")).toBeTruthy();
+        expect(screen.getByRole("button", { name: "Search" })).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/components/SearchForm.test.tsx]: 対応しました。screen.getBy... は要素がない場合に throw するため、render テストでは冗長な toBeTruthy() を削除し、存在確認の呼び出しだけにしました。検証: pnpm --filter @paper-tools/web test -- src/components/SearchForm.test.tsx / git diff --check

File: packages/web/src/components/SearchForm.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/components/SearchForm.test.tsx:32]: 対応しました。送信テストの入力を "  test query  " に変更し、maxResults は既定値を user.clear してから 50 を入力することで、onSearchMock が "test query", 50 で呼ばれることを確認しています。検証: pnpm --filter @paper-tools/web test -- src/components/SearchForm.test.tsx / git diff --check
